### DISCUSSION TITLE
Cant use GTK+ as Embedded XRDP Kiosk

### DISCUSSION THREAD

**Discussion by CE1CECL** [neutrinolabs/xrdp#2356] [Q&A]
Hi there! I cannot use XRDP with GTK programs, specifically, QEMU & virt-viewer without a Desktop Environment loaded up. I changed my startwm.sh script to run qemu, with gtk and it doesn't go to full screen, nor can i minimize or maximize the window, they do nothing. SDL works but keyboard and mouse don't, GTK  was the best one so far. Any suggestions? Thanks (see attached for what I mean)
![Capture](https://user-images.githubusercontent.com/52040854/188327564-2598c43d-264a-4d8c-9af3-ccbcc5bb3410.PNG)

**Comment by matt335672** [comment]
@CE1CECL - If you want to resize an X11 application, you'll need to run at least a basic window manager with it.

You could start with twm if you're familiar with it - it's really basic, but you'll get the idea. Your startwm.sh will need to be something like:-

```
#!/bin/bash

<some stuff>

twm&
WM_PID=$!

my_application
STATUS=$?

kill $WM_PID
exit $STATUS
```
